pkg/background/update_request_controller.go

```diff
@@ -179,15 +172,25 @@ func (c *Controller) syncUpdateRequest(key string) error {
 	if err != nil {
 		return err
 	}
-	ur, err := c.urLister.Get(urName)
+	// if state is not set, try to set it to pending
+	{
+		ur, err := c.urLister.Get(urName)
+		if err != nil {
+			return err
+		}
+		if ur.Status.State == "" {
+			ur = ur.DeepCopy()
+			ur.Status.State = kyvernov1beta1.Pending
+			_, err := c.kyvernoClient.KyvernoV1beta1().UpdateRequests(config.KyvernoNamespace()).UpdateStatus(context.TODO(), ur, metav1.UpdateOptions{})
+			// in any case we want to return and wait the next reconcile
+			return err
+		}
+	}
+	ur, ok, err := c.markUR(urName)
 	if err != nil {
 		if apierrors.IsNotFound(err) {
 			return nil
 		}
```

Comment:
This check checks the error returned from `ur, err := c.urLister.Get(urName)`, not `c.markUR(urName)` right?

Reply:
Yep

Reply:
I hope so...